ingestors/kafka/src/main/scala/hydra/kafka/programs/CreateTopicProgram.scala

```diff
@@ -155,8 +155,11 @@ final class CreateTopicProgram[F[_]: Bracket[*[_], Throwable]: Sleep: Logger] pr
     val td = createTopicRequest.numPartitions.fold(defaultTopicDetails)(numP =>
       defaultTopicDetails.copy(numPartitions = numP.value))
       .copy(partialConfig = defaultTopicDetails.configs ++ getCleanupPolicyConfig)
+
     (for {
-      _ <- Resource.eval(validator.validate(createTopicRequest, topicName, withRequiredFields))
+      metadata <- Resource.eval(metadataAlgebra.getMetadataFor(topicName))
+      createdDate = metadata.map(_.value.createdDate).orElse(Option(createTopicRequest.createdDate))
```

Comment:
Also why `createdDate` is Optional here? Maybe it's better to use `getOrElse` because `metadata.map(_.value.createdDate)` is Optional and `createTopicRequest.createdDate` is default value.

Reply:
`createdDate` is used only in validator. Maybe is better to move it inside validator

Reply:
Done.